shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -1696,13 +1697,178 @@ public static WireCommand readFrom(ByteBufInputStream in, int length) throws IOE
         }
     }
 
+    @Data
+    public static final class GetTableKeys implements Request, WireCommand {
+
+        final WireCommandType type = WireCommandType.GET_TABLE_KEYS;
+        final long requestId;
+        final String segment;
+        final String delegationToken;
+        final int suggestedKeyCount;
+        final ByteBuffer continuationToken; // this used to indicate the point from which the next keys should be fetched.
+
+        @Override
+        public void process(RequestProcessor cp) {
+            cp.getTableKeys(this);
+        }
+
+        @Override
+        public void writeFields(DataOutput out) throws IOException {
+            out.writeLong(requestId);
+            out.writeUTF(segment);
+            out.writeUTF(delegationToken == null ? "" : delegationToken);
+            out.writeInt(suggestedKeyCount);
+            out.writeInt(continuationToken.remaining()); // continuation token length.
+            if (continuationToken.remaining() != 0) {
+                out.write(continuationToken.array(), continuationToken.arrayOffset() + continuationToken.position(), continuationToken.remaining());
+            }
+        }
+
+        public static WireCommand readFrom(ByteBufInputStream in, int length) throws IOException {
+            long requestId = in.readLong();
+            String segment = in.readUTF();
+            String delegationToken = in.readUTF();
+            int suggestedKeyCount = in.readInt();
+            int dataLength = in.readInt();
+            byte[] continuationToken = new byte[dataLength];
```

Comment:
Check that dataLength is less than length so that we don't try to read off the end.

Reply:
done.